specs/activity-api.yaml

```diff
@@ -167,25 +135,23 @@ paths:
                 type: string
                 format: binary
         404:
-          description: Not Found
-          content:
-            application/json:
-              schema:
-                $ref: 'common.yaml#/schemas/ErrorMessage'
+          $ref: 'common.yaml#/responses/NotFound'
         500:
-          description: Server Error
-          content:
-            application/json:
-              schema:
-                $ref: 'common.yaml#/schemas/ErrorMessage'
+          $ref: 'common.yaml#/responses/ServerError'
 
   /activity/{activityId}/exec/{batchId}:
     get:
       tags:
         - requestor_control
       summary: Queries for ExeScript batch results.
-      description: '**Note:** This call shall collect ExeScriptCommand result objects
-        received directly from ExeUnit.'
+      description: 'This call shall collect ExeScriptCommand result objects
+        received directly from ExeUnit (via the long polling pattern).
+        The events are persistent, ie. the API can be called repeatedly
```

Comment:
Is it really a good idea for the deamon to store command output in the database? IMHO that should be left for the requestor to decide.

Reply:
The track record of exec events received from the exeunit is important evidence, eg. for conflict reconciliation. This is why we agreed to align the "commercially binding event" APIs (agreement events, payment events and also activity exec events) to be persistent, so that the evidence collection does not depend on the requestor/provider agent applications.

Reply:
I can imagine the unification of events, it just seems impractical in this case (and that was the initial plan).

- currently there is no conflict reconciliation mechanism
- a policy of sweeping command output from the database has not been considered; the database may grow indefinitely and will degrade in performance

additionally
- apps may output binary data
- conflict reconciliation mechanism should use output digests instead of outputs

Reply:
@stranger80 This is a bad idea. Please reconsider your position on the subject for the following reasons:

- command output has no evidence value since its origin and content cannot be verified (e.g. no provider's signature)
- the reconciliation mechanism must not use raw output (potentially GBs of data) otherwise I'd question its expediency
- applications are free to output binary files to stdout / stderr, not just text
- daemon was not designed to support heavy database workloads
- daemon's DB was not designed to efficiently store GBs of data (sqlite)
- data retention policy is a more complex subject than this issue

if you don't agree, please discuss it on a call. For now, I'm willing to implement this as an optional daemon feature.

Reply:
I take your points, @mfranciszkiewicz . Let's leave persistence out of scope.